patches/server/1045-Vault-API.patch

```diff
@@ -0,0 +1,305 @@
++        Preconditions.checkArgument(deactivationRange >= config.activationRange(), "DeactivationRange must be greater or equal to ActivationRange");
++
++        config = new VaultConfig(config.lootTable(), config.activationRange(), deactivationRange, config.keyItem(), config.overrideLootTableToDisplay());
++        this.getSnapshot().setConfig(config);
++    }
++
++    @Override
++    public Collection<Player> getRewardedPlayers() {
++        ImmutableSet.Builder<Player> players = ImmutableSet.builder();
++
++        for (UUID uuid : this.getTileEntity().getServerData().getRewardedPlayers()) {
++            Player player = Bukkit.getPlayer(uuid);
++            if (player != null) {
++                players.add(player);
++            }
++        }
++        return players.build();
++    }
++
++    @Override
++    public Collection<UUID> getRewardedPlayerUUIDs() {
++        return this.getTileEntity().getServerData().getRewardedPlayers();
++    }
++
++    @Override
++    public boolean hasRewardedPlayer(final Player player) {
++        Preconditions.checkArgument(player != null, "Player cannot be null");
++
++        return this.getTileEntity().getServerData().hasRewardedPlayer(((CraftPlayer) player).getHandle());
++    }
++
++    @Override
++    public boolean hasRewardedPlayer(final UUID uniqueId) {
++        Preconditions.checkArgument(uniqueId != null, "UUID cannot be null");
++
++        return this.getTileEntity().getServerData().hasRewardedPlayer(uniqueId);
++    }
++
++    @Override
++    public void addToRewardedPlayers(Player player) {
++        Preconditions.checkArgument(player != null, "Player cannot be null");
++
++        this.getTileEntity().getServerData().addToRewardedPlayers(((CraftPlayer) player).getHandle());
++    }
++
++    @Override
++    public void addToRewardedPlayers(final UUID uniqueId) {
++        Preconditions.checkArgument(uniqueId != null, "UUID cannot be null");
++
++        this.getTileEntity().getServerData().addToRewardedPlayers(uniqueId);
++    }
++
++    @Override
++    public void removeFromRewardedPlayers(Player player) {
++        Preconditions.checkArgument(player != null, "Player cannot be null");
++
++        this.getTileEntity().getServerData().getRewardedPlayers().remove(((CraftPlayer) player).getHandle());
++        this.getTileEntity().getServerData().markChanged();
++    }
++
++    @Override
++    public void removeFromRewardedPlayers(final UUID uniqueId) {
++        Preconditions.checkArgument(uniqueId != null, "UUID cannot be null");
++
++        this.getTileEntity().getServerData().getRewardedPlayers().remove(uniqueId);
++        this.getTileEntity().getServerData().markChanged();
++    }
++
++    @Override
++    public List<ItemStack> getItemsToEject() {
++        return this.getTileEntity().getServerData().getItemsToEject().stream().map(i -> i.asBukkitCopy()).toList();
++    }
++
++    @Override
++    public void setItemsToEject(final List<ItemStack> itemsToEject) {
```

Comment:
This doesn't check empty item, so i can easily create a corrupted block entity

Reply:
Empty items seems to be checked internally and handled so null and AIR items should work fine ithink
Unless you want me to explicitly mark ItemStack as notnull here
![image](https://github.com/user-attachments/assets/9803c317-f08b-459d-a457-10abb48a1ff6)

Reply:
The codec doesn't support those empty item.
For reference here is the stacktrace when you save the world without interacting with the vault (or using the data command): https://pastes.dev/faeBdNZWcj